core/src/main/java/org/apache/calcite/sql/SqlIfStmt.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql;
+
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.util.ImmutableNullableList;
+
+import java.util.List;
+/**
+ * Parse tree for a {@code SqlIfStmt}.
+ */
+public class SqlIfStmt extends SqlCall {
+  public static final SqlSpecialOperator OPERATOR =
+      new SqlSpecialOperator("IF", SqlKind.CONDITIONAL_STATEMENT);
+
+  public final SqlNodeList conditionalStmtListPairs;
+  public final SqlNodeList elseStmtList;
+
+  /**
+   * Creates a {@code SqlTablePartitionByColumn}.
+   * @param pos                       Parser position, must not be null.
+   * @param conditionalStmtListPairs  List of conditional expression pairs
+   *                                  with StatementList.
+   * @param elseStmtList              List of statement in the else clause.
+   */
+  public SqlIfStmt(final SqlParserPos pos,
+      final SqlNodeList conditionalStmtListPairs,
+      final SqlNodeList elseStmtList) {
+    super(pos);
+    this.conditionalStmtListPairs = conditionalStmtListPairs;
```

Comment:
nit: Should we assert that this is non-null?

Reply:
It cannot be null, base on the logic in the function IfStmt() in parserImpls.ftl. It would at least contain "IF <condition> THEN <statemment list>" part.

Reply:
That might be true for now. But isn't that making an assumption that the caller (e.g. the parser) never changes the way they construct this class? It's usually best practice to assert that it's non-null here.

Reply:
(This suggestion applies to all constructors added in this PR)

Reply:
Sorry about the notification for this comment got ignored. Fixed in https://github.com/googleinterns/calcite/pull/245/commits/bc902df2d7f4882337c55eac53aa8dc3e7a31286.

Reply:
nit: you can have the Objects.requireNonNull() call directly in the assignment like this (rather than on a separate line):
this.conditionalStmtListPairs = Objects.requireNonNull(conditionalStmtListPairs)
You could also do this for elseStmtList below.